app/controllers/api_controller/providers.rb

```diff
@@ -132,13 +141,37 @@ def validate_credential_attributes(provider, creds)
                                     invalid_attrs.join(', '), provider.class.name, auth_attrs.join(", "))
     end
 
+    def check_connection_configurations(data)
+      configurations = data["connection_configurations_array"]
+      return unless configurations
+      raise BadRequestError, "Attribute connection_configurations_array must be an Array" unless
+        configurations.kind_of?(Array)
+    end
+
+    def merge_connection_configurations(provider, data)
+      provider_configurations = provider_connection_configurations(provider)
+      data_configurations = data_connection_configurations(data.deep_symbolize_keys)
+
+      provider_configurations.deep_merge(data_configurations.symbolize_keys).map { |_, v| v }
+    end
+
+    def provider_connection_configurations(provider)
+      return {} unless provider.respond_to?(:connection_configurations_array)
```

Comment:
Can any provider not respond, despite your addition `has_many :connection_configurations_array ...` in Provider?